### PR TITLE
7903187: jextract should support --version option

### PR DESCRIPTION
**Pull request by sundararajana**
added --version option that prints the underlying Java version and exits.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no reviews required)

### Issue
 * [CODETOOLS-7903187](https://bugs.openjdk.java.net/browse/CODETOOLS-7903187): jextract should support --version option

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jextract pull/31/head:pull/31` \
`$ git checkout pull/31`

Update a local copy of the PR: \
`$ git checkout pull/31` \
`$ git pull https://git.openjdk.java.net/jextract pull/31/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 31`

View PR using the GUI difftool: \
`$ git pr show -t 31`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jextract/pull/31.diff">https://git.openjdk.java.net/jextract/pull/31.diff</a>

</details>

### CODE REVIEW

File: src/main/java/org/openjdk/jextract/JextractTool.java [hunk 1]
```diff
@@ -351,6 +352,11 @@ private int run(String[] args) {
             return OPTION_ERROR;
         }
 
+        if (optionSet.has("--version")) {
+            err.println(System.getProperty("java.version"));
```
Comment (mcimadamore) [inline src/main/java/org/openjdk/jextract/JextractTool.java:356]: looking at other tools such as `javac` and `javadoc` they print out the tool name and then the version - e.g. `javac 15`. We should at least be consistent with that.

Then there `jtreg` which prints out a lot more info:

```
jtreg 6.2+1-12
Installed in /opt/jtreg/lib/jtreg.jar
Running on platform version 11.0.15 from /usr/lib/jvm/java-11-openjdk-amd64.
Built with Java(TM) 2 SDK, Version 1.8.0_261-b33 on January 22, 2022.
Copyright (c) 1999, 2022, Oracle and/or its affiliates. All rights reserved.
Use is subject to license terms.
JT Harness, version 6.0 ea b14 (November 04, 2020)
Java Assembler Tools, version 7.0 ea b09 (June 04, 2020)
TestNG (testng.jar): version 7.3.0
TestNG (jcommander.jar): version unknown
TestNG (guice.jar): version 4.2.3
JUnit (junit.jar): version 4.13.2
JUnit (hamcrest.jar): version 2.2
```

While we don't to replicate what jtreg does 100%, a couple of comments:

* adding the specific version with which jextract was built could not be a bad idea (for debugging)
* adding at least the LLVM version used could also be good

**Comment by mcimadamore** [comment]
Ouch - I didn't spot that the README file needs to be updated to document the new option (since we give a list of all the supported options).

**Comment by sundararajana** [comment]
I'll piggyback adding --version to README.md in the next change.

**Pull request closed without merging**